### PR TITLE
BigQueryDynamicTableSource junit test cases added

### PR DESCRIPTION
**Pull request by Sridhar-Durgam2022**

### CODE REVIEW

File: src/test/java/com/google/cloud/flink/bigquery/BigQueryDynamicTableSourceTest.java [hunk 1]
```diff
@@ -0,0 +1,174 @@
...
+    assertThat(bigQueryDynamicTableSource instanceof BigQueryDynamicTableSource);
+    assertThat(bigQueryDynamicTableSource.getChangelogMode()).isNotNull();
+    assertThat(bigQueryDynamicTableSource.getChangelogMode() instanceof ChangelogMode);
+    assertThat(bigQueryDynamicTableSource.copy() instanceof BigQueryDynamicTableSource);
+    assertThat(bigQueryDynamicTableSource.asSummaryString()).isEqualTo("BigQuery Table Source");
+    assertThat(resultScanRuntimeProvider).isNotNull();
+  }
+
+  @Test
+  public void testSourceFunction() throws Exception {
```
Comment (VinayLondhe14) [inline src/test/java/com/google/cloud/flink/bigquery/BigQueryDynamicTableSourceTest.java:101]: Why are we testing the source function here?

File: src/test/java/com/google/cloud/flink/bigquery/BigQueryDynamicTableSourceTest.java [hunk 2]
```diff
@@ -0,0 +1,174 @@
...
+
+    ArrayList<String> readStreamNames = new ArrayList<String>();
+
+    // Initialize the constructor
+    BigQueryDynamicTableSource bigQueryDynamicTableSource =
+        new BigQueryDynamicTableSource(
+            decodingFormat, producedDataType, readStreamNames, mockBigQueryClientFactory);
+
+    ScanRuntimeProvider resultScanRuntimeProvider =
+        bigQueryDynamicTableSource.getScanRuntimeProvider(new MockScanContext());
```
Comment (VinayLondhe14) [inline src/test/java/com/google/cloud/flink/bigquery/BigQueryDynamicTableSourceTest.java:90]: Why cant we use the Mockito library to mock the Context object?

Comment (Sridhar-Durgam2022) [reply src/test/java/com/google/cloud/flink/bigquery/BigQueryDynamicTableSourceTest.java]: Created mock object for scan context

File: src/test/java/com/google/cloud/flink/bigquery/BigQueryDynamicTableSourceTest.java (hunk 1 above)
Comment (Sridhar-Durgam2022) [reply src/test/java/com/google/cloud/flink/bigquery/BigQueryDynamicTableSourceTest.java]: removed as not needed here

File: pom.xml [hunk 3]
```diff
@@ -64,6 +64,8 @@
         <jackson-databind.version>2.13.2.2</jackson-databind.version>               
         <jsqlparser.version>3.1</jsqlparser.version>
         <junit.version>4.13.1</junit.version>
+        <mockito-core.version>3.10.0</mockito-core.version>
```
Comment (davidrabinowitz) [inline pom.xml:67]: Mockito should have one version (4.5.1)

**Review by davidrabinowitz — CHANGES_REQUESTED** [review]

File: src/test/java/com/google/cloud/flink/bigquery/BigQueryDynamicTableSourceTest.java [hunk 4]
```diff
@@ -0,0 +1,157 @@
...
+public class BigQueryDynamicTableSourceTest {
+
+  static String bigqueryReadTable;
+  static ObjectIdentifier tableIdentifier;
+  static TableSchema tableSchema;
+  static Configuration options = new Configuration();
+
+  @BeforeClass
+  public static void setup() {
+    bigqueryReadTable = "q-gcp-6750-pso-gs-flink-22-01.wordcount_dataset.wordcount_output";
```
[the remark this answers is not in the document]
Comment (Sridhar-Durgam2022) [reply src/test/java/com/google/cloud/flink/bigquery/BigQueryDynamicTableSourceTest.java]: changed to dummy table name for testing

File: pom.xml (hunk 3 above)
Comment (Sridhar-Durgam2022) [reply pom.xml]: changed the version

**Pull request closed without merging**